src/main/java/org/folio/services/validator/engine/ValidationEngineServiceImpl.java

```diff
@@ -162,23 +164,22 @@ private void validatePasswordByRexExpRule(final String password,
     Promise<JsonObject> promise = Promise.promise();
     String okapiUrl = headers.get(OKAPI_URL_HEADER);
     String userNameRequestUrl = String.format("%s/users?query=id==%s", okapiUrl, userId);
-    HttpClientRequest request = httpClient.getAbs(userNameRequestUrl);
+    HttpRequest<Buffer> request = webClient.getAbs(userNameRequestUrl);
     request
       .putHeader(OKAPI_HEADER_TOKEN, headers.get(OKAPI_HEADER_TOKEN))
       .putHeader(OKAPI_HEADER_TENANT, headers.get(OKAPI_HEADER_TENANT))
       .putHeader(HttpHeaders.CONTENT_TYPE.toString(), MediaType.APPLICATION_JSON)
       .putHeader(HttpHeaders.ACCEPT.toString(), MediaType.APPLICATION_JSON)
-      .handler(response -> {
-        if (response.statusCode() != 200) {
-          response.bodyHandler(buf -> {
+      .send(ar -> {
+        if (ar.succeeded()) {
+          HttpResponse<Buffer> response = ar.result();
+          if (response.statusCode() != 200) {
             String message = "Error looking up user at url '" + userNameRequestUrl
               + "' Expected status code 200, got '" + response.statusCode() +
-              "' :" + buf.toString();
+              "' :" + response.bodyAsString();
             promise.fail(message);
```

Comment:
Converting this into a guard clause allows to drop the else clause and improves readability:

    if (response.statusCode() != 200) {
      return;
    }